momepy/functional/_diversity.py

```diff
@@ -302,3 +329,44 @@ def describe_reached(
     result.index.names = result_index.names
 
     return result
+
+
+def percentile(
+    y: Series,
+    graph: Graph,
+    q: tuple | list = [25, 50, 75],
+):
+    """Calculates linearly weighted percentiles of ``y`` values using
+    the neighbourhoods and weights defined in ``graph``.
+
+    The specific interpolation method implemented is "hazen".
```

Comment:
What does that mean?

Reply:
its one of the interpolation method in np.interpolate, and the one implemented in momepy.Percentile(weights='linear_weighted').
Basically we divide the weights/xs by their sum .